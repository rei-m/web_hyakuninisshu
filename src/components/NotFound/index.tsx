import * as React from 'react';
import styled from 'styled-components';
import { appTheme, withAppTheme } from '@src/styles';
import Dogeza from '@src/components/Dogeza';

const Root = withAppTheme(styled.div)`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${({ theme }) => theme.colorThin};
  flex-direction: column;
  box-sizing: border-box;
  min-height: calc(100vh - ${({ theme }) => theme.headerHeight});

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    min-height: calc(100vh - ${({ theme }) => theme.headerHeightWide});
  }
`;

const Message = styled.div`
  font-size: 1.8rem;
`;

const NotFound: React.SFC = () => (
  <Root>
    <Message>
      ページが見つかりませんでした。<br />トップページにお戻りください。
    </Message>
    <Dogeza
      alt="ページが見つかりませんでした"
      style={{
        marginTop: appTheme.spacing4x
      }}
    />
  </Root>
);

export default NotFound;
