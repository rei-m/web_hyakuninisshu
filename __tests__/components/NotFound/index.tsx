import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import NotFound from '../../../src/components/NotFound';

describe('<NotFound />', () => {
  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<NotFound />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
