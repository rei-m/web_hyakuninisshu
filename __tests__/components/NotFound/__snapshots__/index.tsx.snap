// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render component 1`] = `
<div
  className="sc-bwzfXH eCHUJQ"
>
  <div
    className="sc-htpNat fSNmNw"
  >
    ページが見つかりませんでした。
    <br />
    トップページにお戻りください。
  </div>
  <img
    alt="ページが見つかりませんでした"
    className="sc-bdVaJa dxKQNh"
    src="dogeza_businessman.png"
    style={
      Object {
        "marginTop": "32px",
      }
    }
  />
</div>
`;
